soroban-sdk-auth/src/lib.rs

```diff
@@ -8,13 +8,48 @@ use crate::public_types::{
     SignaturePayloadV0,
 };
 
+/// Users of this module will need to pass a struct that implements NonceAuth
+/// NonceAuth should manage a nonce in the contracts storage
+///
+///
+/// Example implementation for NonceAuth -
```

Comment:
It's customary to put these in an examples section.
```suggestion
/// ### Examples
```

Reply:
done